tests/dGameTests/dComponentsTests/DestroyableComponentTests.cpp

```diff
@@ -41,10 +41,19 @@ class DestroyableTest : public GameDependenciesTest {
 TEST_F(DestroyableTest, DestroyableComponentSerializeConstructionTest) {
 	destroyableComponent->Serialize(&bitStream, true, flags);
 	// Assert that the full number of bits are present
-	ASSERT_EQ(bitStream.GetNumberOfUnreadBits(), 460);
+	ASSERT_EQ(bitStream.GetNumberOfUnreadBits(), 748);
 	{
 		// Now read in the full serialized construction BitStream
-		bool optionStatusImmunityInfo{}; // Values under this option are unused.
+		bool optionStatusImmunityInfo{};
+		uint32_t ImmuneToBasicAttackCount{};
+		uint32_t ImmuneToDamageOverTimeCount{};
+		uint32_t ImmuneToKnockbackCount{};
+		uint32_t ImmuneToInterruptCount{};
+		uint32_t ImmuneToSpeedCount{};
+		uint32_t ImmuneToImaginationGainCount{};
+		uint32_t ImmuneToImaginationLossCount{};
+		uint32_t ImmuneToQuickbuildInterruptCount{};
+		uint32_t ImmuneToPullToPointCount{};
```

Comment:
it may be worth considering pushing some immune stacks and checking that the right variables are set in the right spots i.e. basicAttackCount set to 1, DamageOverTImeCount set to 2 etc. so if the order is swapped on accident, this test would catch that

Reply:
Can do sir